std/typecons.d

```diff
@@ -3290,14 +3290,118 @@ unittest
 }
 
 /**
+ * Determines if the $(D Source) type satisfies all interface requirements of
+ * $(D Targets)
+ */
+private template isConceptWith(Source, Targets...)
+if (Targets.length >= 1 && allSatisfy!(isMutable, Targets))
+{
+    import std.typetuple : staticMap;
+
+    // strict upcast
+    auto isConceptWith()() @trusted pure nothrow
```

Comment:
Functions and function templates can be in the same overload set now. Also, this function is simply `@safe`.